src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs

```diff
@@ -103,26 +104,30 @@ private static int GetTypeLength(string input, int offset, out StringSegment typ
                 return 0;
             }
 
-            // Parse the type, i.e. <type> in media type string "<type>/<subtype>; param1=value1; param2=value2"
-            var typeLength = HttpTokenParsingRules.GetTokenLength(input, offset);
+            // ??? GetWhitespaceLength() incorrectly stops if it hits a comment. Should either use or remove the
+            // ??? HttpTokenParsingRules comment support. File a bug?
+            var current = offset + HttpTokenParsingRules.GetWhitespaceLength(input, offset);
```

Comment:
Because nothing else throws away leading whitespace in the `string` passed to the `MediaType` constructor. Spaces are just as valid there as they are between any other tokens and delimiters.